src/test/java/htsjdk/variant/vcf/VCFHeaderLineTranslatorUnitTest.java

```diff
@@ -46,17 +50,116 @@ public void testParseVCF4HeaderLine() {
         }
 
         // test with an escaped quote at the end
         try {
-            final Map<String, String> values7 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2, "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it\\\">", null);
+            final Map<String,String> values7 = VCFHeaderLineTranslator.parseLine(VCFHeaderVersion.VCF4_2, "<ID=ANNOTATION,Description=\"ANNOTATION \\n with a newline in it\\\">", null);
             Assert.fail("Should have thrown a TribbleException for having an unclosed quote in the description line");
-        } catch (TribbleException.InvalidHeader e) {
         }
+        catch (TribbleException.InvalidHeader e) {
+        }
+
+    }
+
+    @DataProvider(name = "validHeaderLines")
+    private Object[][] getValidHeaderLines() {
+        List<String> idDesc = Arrays.asList("ID", "Description");
+        List<Object> none = Collections.emptyList();
+        List<Object> sourceVersion = Arrays.asList("Source", "Version");
+        List<String> extra = Arrays.asList("Extra");
+        return new Object[][]{
+                // to parse, expected, recommended
+                {"<ID=X,Description=\"Y\",Source=\"source\",Version=\"1.2.3\">", idDesc, sourceVersion},
+                {"<ID=X,Description=\"Y\",Source=\"source\">", idDesc, sourceVersion},
+                {"<ID=X,Description=\"Y\",Version=\"1.2.3\">", idDesc, sourceVersion},
+                {"<ID=X,Description=\"Y\">", idDesc, sourceVersion},
+                {"<ID=X>", idDesc, sourceVersion},
+                {"<ID=X,Description=\"Y\",Extra=\"extra\",Source=\"source\",Version=\"1.2.3\">", idDesc, sourceVersion},
+
+                {"<ID=X>", idDesc, none},
+                {"<ID=X,Description=\"Y\">", idDesc, none},
+                {"<ID=X,Description=\"Y\",Extra=\"extra\">", idDesc, none},
+                {"<ID=X,Description=<Y>>", idDesc, none},
+                {"<ID=X,Description=\"Y\",Extra=E>", idDesc, none},
+                
+                {"<ID=X,Description=\"Y\",Extra=\"extra\">", idDesc, extra},
+                {"<ID=X,Description=\"Y\">", idDesc, extra},
+                {"<>", none, none},
+                {"<>", none, extra},
+                {"<>", none, sourceVersion}
+        };
+    }
+    
+    @DataProvider(name = "invalidHeaderLines")
+    private Object[][] getInvalidHeaderLines() {
+        List<String> idDesc = Arrays.asList("ID", "Description");
+        List<Object> none = Collections.emptyList();
+        List<String> sourceVersion = Arrays.asList("Source", "Version");
+        return new Object[][]{
+                // to parse, expected, recommended, error message
+                {"<Description=\"Y\",ID=X>", idDesc, none, "Tag Description in wrong order (was #1, expected #2)"},
+                {"<ID=X,Desc=\"Y\">", idDesc, none, "Unexpected tag Desc"},
+                {"<>", idDesc, none, "Header with no tags is not supported when there are expected tags"},
```

Comment:
That is not my experience.
I reverted that switch back and the tests still pass.
The main change to the switch was 
a) formatting 
b) handling of embedded <> symbols - (which I do not actually test for, yikes)
c) handling of the empty case <> - (again I do not test for it, the old code will create a map with null -> null entry, the new code created an empty map)

Reply:
@romanzenka I didn't mean for you to remove all of the new tests, only the one specific *test case* I commented on, because it will no longer pass with the switch statement reverted. All the rest of the tests are a welcome addition.

Reply:
Sorry, my misunderstanding. You made it sound like some other test cases cover this already.

Reply:
Pushed the cases back, made sure they pass